geode-core/src/test/java/org/apache/geode/TX2JUnitTest.java

```diff
@@ -0,0 +1,217 @@
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode;
+
+import static org.apache.geode.distributed.ConfigurationProperties.MCAST_PORT;
+import static org.junit.Assert.*;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Iterator;
+
+import org.apache.geode.cache.Cache;
+import org.apache.geode.cache.CacheFactory;
+import org.apache.geode.cache.CacheTransactionManager;
+import org.apache.geode.cache.Region;
+import org.apache.geode.cache.RegionFactory;
+import org.apache.geode.cache.RegionShortcut;
+import org.apache.geode.distributed.internal.DistributionConfig;
+import org.apache.geode.internal.cache.GemFireCacheImpl;
+import org.apache.geode.internal.cache.TXManagerImpl;
+import org.apache.geode.internal.cache.TXStateProxyImpl;
+import org.apache.geode.internal.logging.LogService;
+import org.apache.geode.test.junit.categories.IntegrationTest;
+import org.apache.logging.log4j.Logger;
+import org.junit.After;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.contrib.java.lang.system.RestoreSystemProperties;
+import org.junit.experimental.categories.Category;
+
+@Category(IntegrationTest.class)
+public class TX2JUnitTest {
+
+  private static final Logger logger = LogService.getLogger();
+
+  @Rule
+  public RestoreSystemProperties restoreSystemProperties = new RestoreSystemProperties();
+
+  long k1 = 1;
+  long k2 = 2;
+  long k3 = 3;
+  long k4 = 4;
+  String v1 = "value1";
+  String v2 = "value2";
+  String v3 = "value3";
+  String regionName = "region1";
+  Cache cache;
+
+  enum SetOp {
+    KEYSET, VALUES, ENTRYSET;
+  }
+
+  @After
+  public void tearDownTXJUnitTest() throws Exception {
+    closeCache();
+  }
+
+  @Test
+  public void testValuesCallStartsTx() {
+    verifySetOp(SetOp.VALUES, false);
+  }
+
+  @Test
+  public void testKeySetCallStartsTx() {
+    verifySetOp(SetOp.KEYSET, false);
+  }
+
+  @Test
+  public void testEntrySetCallStartsTx() {
+    verifySetOp(SetOp.ENTRYSET, false);
+  }
+
+  @Test
+  public void testValuesCallNotStartTxIfDisabled() {
+    verifySetOp(SetOp.VALUES, true);
+  }
+
+  @Test
+  public void testKeySetCallNotStartTxIfDisabled() {
+    verifySetOp(SetOp.KEYSET, true);
+  }
+
+  @Test
+  public void testEntrySetCallNotStartTxIfDisabled() {
+    verifySetOp(SetOp.ENTRYSET, true);
+  }
+
+  private void verifySetOp(SetOp op, boolean preventSetOpToStartTx) {
```

Comment:
You could probably use JUnitParams or JUnit Parameterized in this test so that you write a single test method and then provide it with different parameter (SetOps and boolean) pairs.

Search through Geode tests for "`@RunWith(Parameterized.class)`" and "`@RunWith(JUnitParamsRunner.class)`" to see examples of both approaches.

Reply:
StatArchiveHandlerIntegrationTest.java is a decent example of using JUnitParams with `@Parameters` on the test method specifying groups of parameters.

Reply:
Thanks for the reference. I made the suggested change.